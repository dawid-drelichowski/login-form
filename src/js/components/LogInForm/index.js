import React, {Component} from 'react'
import ValidationAwareInput from 'components/ValidationAwareInput'

export default class LogInForm extends Component {
  state = {email: '', password: '', showInvalid: false}
  onInputChange = (id, value) => {
    this.setState({[id]: value})
  }
  onSubmit = event => {
    let state = {showInvalid: true}
    event.preventDefault()

    if (this.state.email === 'test@test.pl' && this.state.password === 'Password1.') { // @todo: login and password should be as hash
      state = {loggedIn: true, showInvalid: false}
      this.props.onLogIn()
    }
    this.setState(state)
  }
  render () {
    return <form method="post" onSubmit={this.onSubmit}>
      <fieldset>
        <ValidationAwareInput type="email" label="E-mail" id="email" name="email" required={true} placeholder="E-mail"
          autoComplete="off" onChange={this.onInputChange}/>
        <ValidationAwareInput type="password" label="Password" id="password" name="password"
          pattern="(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{6,}" required={true} placeholder="Password"
          onChange={this.onInputChange}/>
        <ValidationAwareInput type="checkbox" label="Remember me" id="remember" name="remember"/>
        {this.state.showInvalid ? <div>Invalid email or password</div> : ''}
        <input type="submit" value="Login"/>
      </fieldset>
    </form>
  }
}
