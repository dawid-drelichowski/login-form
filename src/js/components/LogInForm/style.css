.log-in-form {
  background: #fff;
  max-width: 300px;
  padding: 30px;
  margin: 10% auto 0;
  box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
}

.log-in-form .field-set {
  border: 0;
}

.log-in-form .field-set .button {
  text-transform: uppercase;
  background: #4caf50;
  width: 100%;
  border: 0;
  padding: 10px;
  color: #fff;
  font-size: 1rem;
  cursor: pointer;
}

.log-in-form .field-set .invalid {
  color: #c54639;
  font-weight: bold;
  margin-bottom: 10px;
}
