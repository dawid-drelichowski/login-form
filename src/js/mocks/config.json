{
  "email": "dev@dev.pl",
  "password": "$2a$10$omJhm62arfVXzZI5uux67Oa40IlXVTCgWuNzfRInLpps0ap5tiXSG"
}