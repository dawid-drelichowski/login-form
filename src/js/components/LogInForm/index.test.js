import React from 'react'
import LogInForm from './index'
import config from 'mocks/config.json'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('Log in form component', () => {
  function getRenderedComponent (config) {
    return renderer.create(<LogInForm config={config}/>)
  }

  function getShallowComponent (config, onLogIn = () => {}) {
    return shallow(<LogInForm config={config} onLogIn={onLogIn}/>)
  }

  it('should render', () => {
    expect(getRenderedComponent(config).toJSON()).toMatchSnapshot()
  })

  it('should call "onLogIn" callback when the form submitted and login and password valid ', () => {
    const onLogIn = jest.fn(),
      formInstance = getShallowComponent(config, onLogIn).instance()

    formInstance.onInputChange('email', config.email)
    formInstance.onInputChange('password', 'Password666')
    formInstance.onSubmit({preventDefault: jest.fn()})
    expect(onLogIn).toHaveBeenCalledTimes(1)
  })

  it('should show invalid message when login or password invalid', () => {
    const form = getShallowComponent(config),
      instance = form.instance()

    instance.onInputChange('email', 'wrong@wrong.pl')
    instance.onInputChange('password', 'Wrong1')
    form.simulate('submit', {preventDefault: jest.fn()})
    expect(form.contains(<div>Invalid email or password</div>)).toBeTruthy()
  })
})
