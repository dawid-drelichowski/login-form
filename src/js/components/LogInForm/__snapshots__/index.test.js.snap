// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log in form component should render 1`] = `
<form
  className="log-in-form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    className="field-set"
  >
    <div
      className="validation-aware-input"
    >
      <label
        htmlFor="email"
      >
        E-mail
      </label>
      <input
        checked={false}
        className="input"
        id="email"
        name="email"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder="E-mail"
        required={true}
        type="email"
        value=""
      />
      
    </div>
    <div
      className="validation-aware-input"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        checked={false}
        className="input"
        id="password"
        name="password"
        onChange={[Function]}
        onInvalid={[Function]}
        pattern="(?=.*[a-z])(?=.*[A-Z])(?=.*\\\\d).{6,}"
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      
    </div>
    <div
      className="validation-aware-input"
    >
      <label
        htmlFor="remember"
      >
        Remember me
      </label>
      <input
        checked={false}
        className="checkbox"
        id="remember"
        name="remember"
        onChange={[Function]}
        onInvalid={[Function]}
        type="checkbox"
        value=""
      />
      
    </div>
    
    <input
      className="button"
      type="submit"
      value="Login"
    />
  </fieldset>
</form>
`;
